{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "bc97ead6",
   "metadata": {},
   "source": [
    "Загрузите прочтения всех экспериментов из статьи Evaluation of two main RNA-seq approaches for gene quantification in clinical RNA sequencing: polyA+ selection versus rRNA depletion, в которой сравнивают два протокола подготовки библиотек для bulk RNA-Seq: при помощи деплеции рРНК (=total) или при помощи селекции polyA-транскриптов (=polyA+).\n",
    "\n",
    "При выполнении задания используйте Nextflow pipeline."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7693e144",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-09-28 05:31:25--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz\n",
      "Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 130.14.250.10, 130.14.250.12, 2607:f220:41e:250::7, ...\n",
      "Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|130.14.250.10|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 85777118 (82M) [application/x-gzip]\n",
      "Saving to: ‘sratoolkit.3.0.0-ubuntu64.tar.gz’\n",
      "\n",
      "sratoolkit.3.0.0-ub 100%[===================>]  81.80M  9.52MB/s    in 16s     \n",
      "\n",
      "2022-09-28 05:31:41 (5.19 MB/s) - ‘sratoolkit.3.0.0-ubuntu64.tar.gz’ saved [85777118/85777118]\n",
      "\n",
      "sratoolkit.3.0.0-ubuntu64/\n",
      "sratoolkit.3.0.0-ubuntu64/bin/\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/align-info\n",
      "sratoolkit.3.0.0-ubuntu64/bin/align-info.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/bam-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/bam-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cg-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cg-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta\n",
      "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/helicos-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/helicos-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kar\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kar.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/latf-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/latf-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/prefetch\n",
      "sratoolkit.3.0.0-ubuntu64/bin/prefetch.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/prefetch.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/rcexplain\n",
      "sratoolkit.3.0.0-ubuntu64/bin/rcexplain.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sam-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sam-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sam-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-search\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-search.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-stat\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-stat.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srapath\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srapath.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srapath.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srf-load\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srf-load.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/test-sra\n",
      "sratoolkit.3.0.0-ubuntu64/bin/test-sra.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-config\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-config.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate.3\n",
      "sratoolkit.3.0.0-ubuntu64/bin/ncbi/\n",
      "sratoolkit.3.0.0-ubuntu64/bin/ncbi/certs.kfg\n",
      "sratoolkit.3.0.0-ubuntu64/bin/ncbi/default.kfg\n",
      "sratoolkit.3.0.0-ubuntu64/bin/ncbi/vdb-copy.kfg\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srapath-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sratools.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/srf-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/test-sra.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-config.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/abi-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/align-info.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/bam-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/cg-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/fastq-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/helicos-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/illumina-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kar.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/latf-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/prefetch-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/rcexplain.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sam-dump-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-dump.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sff-load.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup-orig.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-search.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-sort.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/bin/sra-stat.3.0.0\n",
      "sratoolkit.3.0.0-ubuntu64/example/\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/base-stats.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/dump-reference.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/gene-lookup.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/mismatch-stats.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/quality-stats.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/simplefastq.pl\n",
      "sratoolkit.3.0.0-ubuntu64/example/perl/splitfastq.pl\n",
      "sratoolkit.3.0.0-ubuntu64/README-blastn\n",
      "sratoolkit.3.0.0-ubuntu64/README-vdb-config\n",
      "sratoolkit.3.0.0-ubuntu64/schema/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/align.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/mate-cache.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/pileup-stats.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/qstat.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/refseq.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/align/seq.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/csra2/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/csra2/csra2.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/csra2/read.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/csra2/reference.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/csra2/stats.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/insdc/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/insdc/insdc.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/insdc/seq.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/insdc/sra.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/clip.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/ncbi.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/pnbrdb.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/seq-graph.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/seq.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/spotname.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/sra.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/stats.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/trace.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/varloc.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/ncbi/wgs-contig.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/454.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/abi.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/generic-fastq.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/helicos.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/illumina.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/ion-torrent.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/nanopore.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/pacbio.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/sra/pevents.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/vdb/\n",
      "sratoolkit.3.0.0-ubuntu64/schema/vdb/built-in.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/schema/vdb/vdb.vschema\n",
      "sratoolkit.3.0.0-ubuntu64/CHANGES\n",
      "sratoolkit.3.0.0-ubuntu64/README.md\n"
     ]
    }
   ],
   "source": [
    "!wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz \n",
    "!tar -xvzf sratoolkit.3.0.0-ubuntu64.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "5d3cf594",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "spots read      : 4,941,237\r\n",
      "reads read      : 9,882,474\r\n",
      "reads written   : 9,882,474\r\n"
     ]
    }
   ],
   "source": [
    "# проверила, все ли скачивается.Прошло без команды с config\n",
    "!sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump SRR3900953"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1632f2ca",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2022-09-28 05:44:50--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
      "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
      "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 10249221 (9.8M) [application/zip]\n",
      "Saving to: ‘fastqc_v0.11.9.zip’\n",
      "\n",
      "fastqc_v0.11.9.zip  100%[===================>]   9.77M  5.31MB/s    in 1.8s    \n",
      "\n",
      "2022-09-28 05:44:52 (5.31 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
      "\n",
      "Archive:  fastqc_v0.11.9.zip\n",
      "  inflating: FastQC/cisd-jhdf5.jar   \n",
      "   creating: FastQC/Configuration/\n",
      "  inflating: FastQC/Configuration/adapter_list.txt  \n",
      "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
      "  inflating: FastQC/Configuration/limits.txt  \n",
      "  inflating: FastQC/fastqc           \n",
      "  inflating: FastQC/fastqc_icon.ico  \n",
      "   creating: FastQC/Help/\n",
      "   creating: FastQC/Help/1 Introduction/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
      "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
      "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
      "   creating: FastQC/Help/2 Basic Operations/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
      "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
      "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
      "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
      "   creating: FastQC/Help/3 Analysis Modules/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
      " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
      "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
      "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
      "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
      "  inflating: FastQC/INSTALL.txt      \n",
      "  inflating: FastQC/jbzip2-0.9.jar   \n",
      "  inflating: FastQC/LICENSE          \n",
      "  inflating: FastQC/LICENSE.txt      \n",
      "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
      "   creating: FastQC/net/\n",
      "   creating: FastQC/net/sourceforge/\n",
      "   creating: FastQC/net/sourceforge/iharder/\n",
      "   creating: FastQC/net/sourceforge/iharder/base64/\n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
      "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
      "   creating: FastQC/org/\n",
      "   creating: FastQC/org/apache/\n",
      "   creating: FastQC/org/apache/commons/\n",
      "   creating: FastQC/org/apache/commons/math3/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/\n",
      "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
      "   creating: FastQC/org/apache/commons/math3/distribution/\n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
      "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
      "   creating: FastQC/org/apache/commons/math3/random/\n",
      "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
      "   creating: FastQC/org/apache/commons/math3/special/\n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
      "   creating: FastQC/org/apache/commons/math3/util/\n",
      "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
      "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
      "  inflating: FastQC/README.md        \n",
      "  inflating: FastQC/README.txt       \n",
      "  inflating: FastQC/RELEASE_NOTES.txt  \n",
      "  inflating: FastQC/run_fastqc.bat   \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  inflating: FastQC/sam-1.103.jar    \r\n",
      "   creating: FastQC/Templates/\r\n",
      "  inflating: FastQC/Templates/fastqc2fo.xsl  \r\n",
      "  inflating: FastQC/Templates/header_template.html  \r\n",
      "   creating: FastQC/Templates/Icons/\r\n",
      " extracting: FastQC/Templates/Icons/error.png  \r\n",
      " extracting: FastQC/Templates/Icons/fastqc_icon.png  \r\n",
      " extracting: FastQC/Templates/Icons/tick.png  \r\n",
      " extracting: FastQC/Templates/Icons/warning.png  \r\n",
      "   creating: FastQC/uk/\r\n",
      "   creating: FastQC/uk/ac/\r\n",
      "   creating: FastQC/uk/ac/babraham/\r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/\r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Help/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Report/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\r\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \r\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \r\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \r\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \r\n",
      " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Results/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \r\n",
      "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \r\n",
      "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \r\n"
     ]
    }
   ],
   "source": [
    "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
    "!unzip fastqc_v0.11.9.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1e6dbbb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Затем сделаем скрипт fastqc выполняемым. \n",
    "!chmod +x FastQC/fastqc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cd855796",
   "metadata": {},
   "source": [
    "NextFlow — это специальный язык (groovy), который позволяет вам запускать целые пайплайны из нескольких последовательно выполняемых программ, следя за тем, чтобы каждая стадия выполнялась строго одна за одной."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "427c5021",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAPSULE: Downloading dependency org.multiverse:multiverse-core:jar:0.7.0wait .. Downloading nextflow dependencies. It may require a few seconds, please wait .. 2/3 KB   \n",
      "CAPSULE: Downloading dependency ch.qos.logback:logback-classic:jar:1.2.9\n",
      "CAPSULE: Downloading dependency org.checkerframework:checker-compat-qual:jar:2.0.0\n",
      "CAPSULE: Downloading dependency org.codehaus.mojo:animal-sniffer-annotations:jar:1.14\n",
      "CAPSULE: Downloading dependency com.google.errorprone:error_prone_annotations:jar:2.1.3\n",
      "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:jar:3.0.10\n",
      "CAPSULE: Downloading dependency com.google.guava:guava:jar:24.1.1-jre\n",
      "CAPSULE: Downloading dependency org.codehaus.jsr166-mirror:jsr166y:jar:1.7.0\n",
      "CAPSULE: Downloading dependency io.nextflow:nf-commons:jar:22.04.5\n",
      "CAPSULE: Downloading dependency org.slf4j:jcl-over-slf4j:jar:1.7.32\n",
      "CAPSULE: Downloading dependency com.beust:jcommander:jar:1.35\n",
      "CAPSULE: Downloading dependency org.slf4j:jul-to-slf4j:jar:1.7.32\n",
      "CAPSULE: Downloading dependency com.google.j2objc:j2objc-annotations:jar:1.1\n",
      "CAPSULE: Downloading dependency ch.grengine:grengine:jar:1.3.0\n",
      "CAPSULE: Downloading dependency jline:jline:jar:2.9\n",
      "CAPSULE: Downloading dependency com.github.zafarkhaja:java-semver:jar:0.9.0\n",
      "CAPSULE: Downloading dependency org.jsoup:jsoup:jar:1.14.3\n",
      "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:jar:3.0.10\n",
      "CAPSULE: Downloading dependency javax.mail:mail:jar:1.4.7\n",
      "CAPSULE: Downloading dependency io.nextflow:nf-httpfs:jar:22.04.5\n",
      "CAPSULE: Downloading dependency commons-lang:commons-lang:jar:2.6\n",
      "CAPSULE: Downloading dependency org.iq80.leveldb:leveldb-api:jar:0.12\n",
      "CAPSULE: Downloading dependency org.codehaus.gpars:gpars:jar:1.2.1\n",
      "CAPSULE: Downloading dependency org.slf4j:slf4j-api:jar:1.7.32\n",
      "CAPSULE: Downloading dependency com.google.code.findbugs:jsr305:jar:1.3.9\n",
      "CAPSULE: Downloading dependency org.objenesis:objenesis:jar:2.1\n",
      "CAPSULE: Downloading dependency com.esotericsoftware.kryo:kryo:jar:2.24.0\n",
      "CAPSULE: Downloading dependency org.apache.ivy:ivy:jar:2.3.0\n",
      "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:jar:3.0.10\n",
      "CAPSULE: Downloading dependency com.googlecode.javaewah:JavaEWAH:jar:1.1.6\n",
      "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:jar:3.0.10\n",
      "CAPSULE: Downloading dependency org.codehaus.groovy:groovy:jar:3.0.10\n",
      "CAPSULE: Downloading dependency org.slf4j:log4j-over-slf4j:jar:1.7.32\n",
      "CAPSULE: Downloading dependency dev.failsafe:failsafe:jar:3.1.0\n",
      "CAPSULE: Downloading dependency javax.activation:activation:jar:1.1.1\n",
      "CAPSULE: Downloading dependency org.yaml:snakeyaml:jar:1.28\n",
      "CAPSULE: Downloading dependency com.jcraft:jsch:jar:0.1.54\n",
      "CAPSULE: Downloading dependency org.pf4j:pf4j:jar:3.4.1\n",
      "CAPSULE: Downloading dependency org.iq80.leveldb:leveldb:jar:0.12\n",
      "CAPSULE: Downloading dependency ch.qos.logback:logback-core:jar:1.2.9\n",
      "CAPSULE: Downloading dependency com.jcraft:jzlib:jar:1.1.1\n",
      "CAPSULE: Downloading dependency com.google.code.gson:gson:jar:2.2.4\n",
      "CAPSULE: Downloading dependency io.nextflow:nextflow:jar:22.04.5\n",
      "CAPSULE: Downloading dependency org.eclipse.jgit:org.eclipse.jgit:jar:5.2.1.201812262042-r\n",
      "CAPSULE: Downloading dependency org.pf4j:pf4j-update:jar:2.3.0\n",
      "CAPSULE: Downloading dependency commons-codec:commons-codec:jar:1.10\n",
      "                                                                      \n",
      "      N E X T F L O W\n",
      "      version 22.04.5 build 5708\n",
      "      created 15-07-2022 16:09 UTC \n",
      "      cite doi:10.1038/nbt.3820\n",
      "      http://nextflow.io\n",
      "\n",
      "\n",
      "Nextflow installation completed. Please note:\n",
      "- the executable file `nextflow` has been created in the folder: /mnt/data/prokaryota/safonnikova/fbb\n",
      "- you may complete the installation by moving it to a directory in your $PATH\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!cd /home/safonnikova/tools/\n",
    "!curl -fsSL https://get.nextflow.io | bash"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e601220",
   "metadata": {},
   "source": [
    "Загружаем прочтение через Nextflow, файл hw1.nf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "39dd541d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting multiqc\n",
      "  Downloading multiqc-1.13-py3-none-any.whl (1.2 MB)\n",
      "\u001b[K     |████████████████████████████████| 1.2 MB 1.6 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting markdown\n",
      "  Downloading Markdown-3.4.1-py3-none-any.whl (93 kB)\n",
      "\u001b[K     |████████████████████████████████| 93 kB 1.2 MB/s  eta 0:00:01\n",
      "\u001b[?25hCollecting simplejson\n",
      "  Downloading simplejson-3.17.6-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (136 kB)\n",
      "\u001b[K     |████████████████████████████████| 136 kB 53.8 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting pyyaml>=4\n",
      "  Downloading PyYAML-6.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (661 kB)\n",
      "\u001b[K     |████████████████████████████████| 661 kB 53.0 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting click\n",
      "  Downloading click-8.1.3-py3-none-any.whl (96 kB)\n",
      "\u001b[K     |████████████████████████████████| 96 kB 2.6 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: requests in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from multiqc) (2.25.1)\n",
      "Collecting spectra>=0.0.10\n",
      "  Downloading spectra-0.0.11.tar.gz (18 kB)\n",
      "Requirement already satisfied: matplotlib>=2.1.1 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from multiqc) (3.5.2)\n",
      "Requirement already satisfied: jinja2>=2.9 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from multiqc) (3.0.2)\n",
      "Collecting lzstring\n",
      "  Downloading lzstring-1.0.4.tar.gz (4.3 kB)\n",
      "Collecting rich-click\n",
      "  Downloading rich_click-1.5.2-py3-none-any.whl (20 kB)\n",
      "Requirement already satisfied: numpy in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from multiqc) (1.22.3)\n",
      "Collecting networkx>=2.5.1\n",
      "  Downloading networkx-2.8.6-py3-none-any.whl (2.0 MB)\n",
      "\u001b[K     |████████████████████████████████| 2.0 MB 48.6 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting rich>=10\n",
      "  Downloading rich-12.5.1-py3-none-any.whl (235 kB)\n",
      "\u001b[K     |████████████████████████████████| 235 kB 59.6 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting future>0.14.0\n",
      "  Downloading future-0.18.2.tar.gz (829 kB)\n",
      "\u001b[K     |████████████████████████████████| 829 kB 49.9 MB/s eta 0:00:01\n",
      "\u001b[?25hCollecting coloredlogs\n",
      "  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)\n",
      "\u001b[K     |████████████████████████████████| 46 kB 3.4 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: MarkupSafe>=2.0 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from jinja2>=2.9->multiqc) (2.0.1)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (2.8.2)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (4.33.3)\n",
      "Requirement already satisfied: pillow>=6.2.0 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (9.1.1)\n",
      "Requirement already satisfied: packaging>=20.0 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (21.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (0.11.0)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (3.0.4)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from matplotlib>=2.1.1->multiqc) (1.4.3)\n",
      "Requirement already satisfied: six>=1.5 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from python-dateutil>=2.7->matplotlib>=2.1.1->multiqc) (1.16.0)\n",
      "Collecting commonmark<0.10.0,>=0.9.0\n",
      "  Downloading commonmark-0.9.1-py2.py3-none-any.whl (51 kB)\n",
      "\u001b[K     |████████████████████████████████| 51 kB 5.2 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: pygments<3.0.0,>=2.6.0 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from rich>=10->multiqc) (2.10.0)\n",
      "Collecting colormath>=3.0.0\n",
      "  Downloading colormath-3.0.0.tar.gz (39 kB)\n",
      "Collecting humanfriendly>=9.1\n",
      "  Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)\n",
      "\u001b[K     |████████████████████████████████| 86 kB 2.6 MB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: importlib-metadata>=4.4 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from markdown->multiqc) (4.8.1)\n",
      "Requirement already satisfied: zipp>=0.5 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from importlib-metadata>=4.4->markdown->multiqc) (3.6.0)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from requests->multiqc) (1.26.6)\n",
      "Requirement already satisfied: chardet<5,>=3.0.2 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from requests->multiqc) (4.0.0)\n",
      "Requirement already satisfied: idna<3,>=2.5 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from requests->multiqc) (2.10)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /home/safonnikova/miniconda3/lib/python3.9/site-packages (from requests->multiqc) (2021.10.8)\n",
      "Building wheels for collected packages: future, spectra, colormath, lzstring\n",
      "  Building wheel for future (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for future: filename=future-0.18.2-py3-none-any.whl size=491059 sha256=92c09126f50a42a1d7fe9a443996c4d126a56f253d2727fcda81c3d36fe74411\n",
      "  Stored in directory: /home/safonnikova/.cache/pip/wheels/2f/a0/d3/4030d9f80e6b3be787f19fc911b8e7aa462986a40ab1e4bb94\n",
      "  Building wheel for spectra (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for spectra: filename=spectra-0.0.11-py3-none-any.whl size=17471 sha256=fe9a982ce3ab399638c08e63279245de1b060b5d8b2d248cd38512f17f6f8359\n",
      "  Stored in directory: /home/safonnikova/.cache/pip/wheels/a4/ef/1c/494fac04a593b2c2c34f5f6b3a0f9a5123a94f484d55f05acb\n",
      "  Building wheel for colormath (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for colormath: filename=colormath-3.0.0-py3-none-any.whl size=39388 sha256=aab5c7b57273d7189f743c360d59f7eabb09a110fe68eb4c39d92e69a5e64fc1\n",
      "  Stored in directory: /home/safonnikova/.cache/pip/wheels/41/fc/b0/725cea6b023e17948a084b1111b4941299859d468c4407e01d\n",
      "  Building wheel for lzstring (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for lzstring: filename=lzstring-1.0.4-py2.py3-none-any.whl size=4575 sha256=c8ab79581b69042c825966f0f68495bf8553016b43d649e9a5e14d8e949f53e3\n",
      "  Stored in directory: /home/safonnikova/.cache/pip/wheels/b2/87/93/fb34b6a6b09b7b4064c4644ffda04bb6fc5273b42c0988e1e8\n",
      "Successfully built future spectra colormath lzstring\n",
      "Installing collected packages: networkx, commonmark, rich, humanfriendly, future, colormath, click, spectra, simplejson, rich-click, pyyaml, markdown, lzstring, coloredlogs, multiqc\n",
      "Successfully installed click-8.1.3 coloredlogs-15.0.1 colormath-3.0.0 commonmark-0.9.1 future-0.18.2 humanfriendly-10.0 lzstring-1.0.4 markdown-3.4.1 multiqc-1.13 networkx-2.8.6 pyyaml-6.0 rich-12.5.1 rich-click-1.5.2 simplejson-3.17.6 spectra-0.0.11\n"
     ]
    }
   ],
   "source": [
    "!pip3 install multiqc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "31e24097",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 22.04.5\n",
      "Launching `./hw1.nf` [clever_chandrasekhar] DSL2 - revision: 3a80d28216\n",
      "\n",
      "  Q U A L I T Y   C O N T R O L  \n",
      "=================================\n",
      "SRA number         : [SRR6410603, SRR6410616]\n",
      "Results location   : /mnt/data/prokaryota/safonnikova/fbb/res_1/results/\n",
      "[-        ] process > DownloadFastQ -\u001b[K\n",
      "[-        ] process > QC            -\u001b[K\n",
      "[-        ] process > MultiQC       -\u001b[K\n",
      "\u001b[4A\n",
      "executor >  local (2)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [ 50%] 1 of 2, cached: 1\u001b[K\n",
      "[e3/2460f1] process > QC (1)            [  0%] 0 of 1\u001b[K\n",
      "[-        ] process > MultiQC           -\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (2)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [ 50%] 1 of 2, cached: 1\u001b[K\n",
      "[e3/2460f1] process > QC (1)            [100%] 1 of 1\u001b[K\n",
      "[-        ] process > MultiQC           -\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (3)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [100%] 2 of 2, cached: 1 ✔\u001b[K\n",
      "[9a/c9d001] process > QC (2)            [ 50%] 1 of 2\u001b[K\n",
      "[-        ] process > MultiQC           -\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (3)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [100%] 2 of 2, cached: 1 ✔\u001b[K\n",
      "[9a/c9d001] process > QC (2)            [100%] 2 of 2 ✔\u001b[K\n",
      "[-        ] process > MultiQC           -\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (4)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [100%] 2 of 2, cached: 1 ✔\u001b[K\n",
      "[9a/c9d001] process > QC (2)            [100%] 2 of 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [  0%] 0 of 1\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (4)\u001b[K\n",
      "[13/e2eae7] process > DownloadFastQ (1) [100%] 2 of 2, cached: 1 ✔\u001b[K\n",
      "[9a/c9d001] process > QC (2)            [100%] 2 of 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [100%] 1 of 1 ✔\u001b[K\n",
      "\u001b[32;1mCompleted at: 28-Sep-2022 09:29:37\n",
      "Duration    : 42m 44s\n",
      "CPU hours   : 1.4 (39.2% cached)\n",
      "Succeeded   : 4\n",
      "Cached      : 1\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!./nextflow run ./hw1.nf --SRA SRR6410603,SRR6410616 -resume"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31481251",
   "metadata": {},
   "source": [
    "Установка Kallisto"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c70f3630",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'kallisto'...\n",
      "remote: Enumerating objects: 4132, done.\u001b[K\n",
      "remote: Counting objects: 100% (146/146), done.\u001b[K\n",
      "remote: Compressing objects: 100% (74/74), done.\u001b[K\n",
      "remote: Total 4132 (delta 76), reused 113 (delta 64), pack-reused 3986\u001b[K\n",
      "Receiving objects: 100% (4132/4132), 4.24 MiB | 1.03 MiB/s, done.\n",
      "Resolving deltas: 100% (2691/2691), done.\n",
      "E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)\n",
      "E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?\n",
      "-- The C compiler identification is GNU 9.0.1\n",
      "-- The CXX compiler identification is GNU 9.0.1\n",
      "-- Check for working C compiler: /usr/bin/cc\n",
      "-- Check for working C compiler: /usr/bin/cc -- works\n",
      "-- Detecting C compiler ABI info\n",
      "-- Detecting C compiler ABI info - done\n",
      "-- Detecting C compile features\n",
      "-- Detecting C compile features - done\n",
      "-- Check for working CXX compiler: /usr/bin/c++\n",
      "-- Check for working CXX compiler: /usr/bin/c++ -- works\n",
      "-- Detecting CXX compiler ABI info\n",
      "-- Detecting CXX compiler ABI info - done\n",
      "-- Detecting CXX compile features\n",
      "-- Detecting CXX compile features - done\n",
      "shared build\n",
      "-- Looking for pthread.h\n",
      "-- Looking for pthread.h - found\n",
      "-- Looking for pthread_create\n",
      "-- Looking for pthread_create - not found\n",
      "-- Looking for pthread_create in pthreads\n",
      "-- Looking for pthread_create in pthreads - not found\n",
      "-- Looking for pthread_create in pthread\n",
      "-- Looking for pthread_create in pthread - found\n",
      "-- Found Threads: TRUE  \n",
      "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\") \n",
      "-- Configuring done\n",
      "-- Generating done\n",
      "-- Build files have been written to: /mnt/data/prokaryota/safonnikova/fbb/kallisto/build\n",
      "\u001b[35m\u001b[1mScanning dependencies of target htslib\u001b[0m\n",
      "[  3%] \u001b[34m\u001b[1mCreating directories for 'htslib'\u001b[0m\n",
      "[  6%] \u001b[34m\u001b[1mNo download step for 'htslib'\u001b[0m\n",
      "[ 10%] \u001b[34m\u001b[1mNo patch step for 'htslib'\u001b[0m\n",
      "[ 13%] \u001b[34m\u001b[1mNo update step for 'htslib'\u001b[0m\n",
      "[ 17%] \u001b[34m\u001b[1mPerforming configure step for 'htslib'\u001b[0m\n",
      "checking for gcc... gcc\n",
      "checking whether the C compiler works... yes\n",
      "checking for C compiler default output file name... a.out\n",
      "checking for suffix of executables... \n",
      "checking whether we are cross compiling... no\n",
      "checking for suffix of object files... o\n",
      "checking whether we are using the GNU C compiler... yes\n",
      "checking whether gcc accepts -g... yes\n",
      "checking for gcc option to accept ISO C89... none needed\n",
      "checking for ranlib... ranlib\n",
      "checking for special C compiler options needed for large files... no\n",
      "checking for _FILE_OFFSET_BITS value needed for large files... no\n",
      "checking shared library type... plain .so\n",
      "checking how to run the C preprocessor... gcc -E\n",
      "checking for grep that handles long lines and -e... /bin/grep\n",
      "checking for egrep... /bin/grep -E\n",
      "checking for ANSI C header files... yes\n",
      "checking for sys/types.h... yes\n",
      "checking for sys/stat.h... yes\n",
      "checking for stdlib.h... yes\n",
      "checking for string.h... yes\n",
      "checking for memory.h... yes\n",
      "checking for strings.h... yes\n",
      "checking for inttypes.h... yes\n",
      "checking for stdint.h... yes\n",
      "checking for unistd.h... yes\n",
      "checking for stdlib.h... (cached) yes\n",
      "checking for unistd.h... (cached) yes\n",
      "checking for sys/param.h... yes\n",
      "checking for getpagesize... yes\n",
      "checking for working mmap... yes\n",
      "checking for gmtime_r... yes\n",
      "checking whether fdatasync is declared... yes\n",
      "checking for fdatasync... yes\n",
      "checking for library containing log... -lm\n",
      "checking for zlib.h... yes\n",
      "checking for inflate in -lz... yes\n",
      "checking for library containing connect... none required\n",
      "configure: WARNING: GCS support not enabled: requires libcurl support\n",
      "configure: WARNING: S3 support not enabled: requires libcurl support\n",
      "configure: creating ./config.status\n",
      "config.status: creating config.mk\n",
      "config.status: creating htslib.pc.tmp\n",
      "config.status: creating config.h\n",
      "[ 20%] \u001b[34m\u001b[1mPerforming build step for 'htslib'\u001b[0m\n",
      "In file included from \u001b[01m\u001b[K./cram/sam_header.h:56\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[K./cram/cram_samtools.h:77\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[K./cram/cram.h:45\u001b[m\u001b[K,\n",
      "                 from \u001b[01m\u001b[Kcram/cram_index.c:65\u001b[m\u001b[K:\n",
      "In function ‘\u001b[01m\u001b[Kkputsn\u001b[m\u001b[K’,\n",
      "    inlined from ‘\u001b[01m\u001b[Kkputs\u001b[m\u001b[K’ at \u001b[01m\u001b[K./htslib/kstring.h:156:9\u001b[m\u001b[K,\n",
      "    inlined from ‘\u001b[01m\u001b[Kcram_index_build\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_index.c:516:9\u001b[m\u001b[K:\n",
      "\u001b[01m\u001b[K./htslib/kstring.h:148:2:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[Kmemcpy\u001b[m\u001b[K’ specified bound 18446744073709551614 exceeds maximum object size 9223372036854775807 [\u001b[01;35m\u001b[K-Wstringop-overflow=\u001b[m\u001b[K]\n",
      "  148 |  \u001b[01;35m\u001b[Kmemcpy(s->s + s->l, p, l)\u001b[m\u001b[K;\n",
      "      |  \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Kcram/cram_io.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Kcram/cram_io.c:2180:27:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.tmp_\u001b[m\u001b[K’ directive writing 5 bytes into a region of size between 1 and 4096 [\u001b[01;35m\u001b[K-Wformat-overflow=\u001b[m\u001b[K]\n",
      " 2180 |      sprintf(path_tmp, \"%s\u001b[01;35m\u001b[K.tmp_\u001b[m\u001b[K%d_%u_%u\", path, pid, thrid, t);\n",
      "      |                           \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Kcram/cram_io.c:2180:24:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kusing the range [0, 4294967295] for directive argument\n",
      " 2180 |      sprintf(path_tmp, \u001b[01;36m\u001b[K\"%s.tmp_%d_%u_%u\"\u001b[m\u001b[K, path, pid, thrid, t);\n",
      "      |                        \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Kcram/cram_io.c:2180:24:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kusing the range [0, 4294967295] for directive argument\n",
      "\u001b[01m\u001b[Kcram/cram_io.c:2180:6:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[Ksprintf\u001b[m\u001b[K’ output between 11 and 4134 bytes into a destination of size 4096\n",
      " 2180 |      \u001b[01;36m\u001b[Ksprintf(path_tmp, \"%s.tmp_%d_%u_%u\", path, pid, thrid, t)\u001b[m\u001b[K;\n",
      "      |      \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "[ 24%] \u001b[34m\u001b[1mNo install step for 'htslib'\u001b[0m\n",
      "[ 27%] \u001b[34m\u001b[1mCompleted 'htslib'\u001b[0m\n",
      "[ 27%] Built target htslib\n",
      "\u001b[35m\u001b[1mScanning dependencies of target kallisto_core\u001b[0m\n",
      "[ 31%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSData.cpp.o\u001b[0m\n",
      "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSTools.cpp.o\u001b[0m\n",
      "[ 37%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Bootstrap.cpp.o\u001b[0m\n",
      "[ 41%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/GeneModel.cpp.o\u001b[0m\n",
      "[ 44%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/H5Writer.cpp.o\u001b[0m\n",
      "[ 48%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Kmer.cpp.o\u001b[0m\n",
      "[ 51%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIndex.cpp.o\u001b[0m\n",
      "[ 55%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIterator.cpp.o\u001b[0m\n",
      "[ 58%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Merge.cpp.o\u001b[0m\n",
      "[ 62%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/MinCollector.cpp.o\u001b[0m\n",
      "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PlaintextWriter.cpp.o\u001b[0m\n",
      "[ 68%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/ProcessReads.cpp.o\u001b[0m\n",
      "[ 72%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PseudoBam.cpp.o\u001b[0m\n",
      "[ 75%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/common.cpp.o\u001b[0m\n",
      "[ 79%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/h5utils.cpp.o\u001b[0m\n",
      "[ 82%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/hash.cpp.o\u001b[0m\n",
      "[ 86%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/main.cpp.o\u001b[0m\n",
      "[ 89%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/weights.cpp.o\u001b[0m\n",
      "[ 93%] \u001b[32m\u001b[1mLinking CXX static library libkallisto_core.a\u001b[0m\n",
      "[ 93%] Built target kallisto_core\n",
      "\u001b[35m\u001b[1mScanning dependencies of target kallisto\u001b[0m\n",
      "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto.dir/main.cpp.o\u001b[0m\n",
      "[100%] \u001b[32m\u001b[1mLinking CXX executable kallisto\u001b[0m\n",
      "[100%] Built target kallisto\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/pachterlab/kallisto.git\n",
    "!apt-get install autoconf\n",
    "!cd kallisto && mkdir build && cd build && cmake .. && make"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b1863ac4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
      "100 64.7M  100 64.7M    0     0  16.7M      0  0:00:03  0:00:03 --:--:-- 16.7M\n"
     ]
    }
   ],
   "source": [
    "# downloading human cDNA (transcriptome)\n",
    "! mkdir index\n",
    "!curl -O ftp://ftp.ensembl.org/pub/release-94/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bf1cde33",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[build] loading fasta file ./index/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
      "[build] k-mer length: 31\n",
      "[build] warning: clipped off poly-A tail (longer than 10)\n",
      "        from 1471 target sequences\n",
      "[build] warning: replaced 5 non-ACGUT characters in the input sequence\n",
      "        with pseudorandom nucleotides\n",
      "[build] counting k-mers ... done.\n",
      "[build] building target de Bruijn graph ...  done \n",
      "[build] creating equivalence classes ...  done\n",
      "[build] target de Bruijn graph has 1118780 contigs and contains 108619921 k-mers \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# creating transcriptome index \n",
    "!./kallisto/build/src/kallisto index -i Homo_sapiens_GRCh38_index.idx ./index/Homo_sapiens.GRCh38.cdna.all.fa.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5354199c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TIMESTAMP          \tDURATION\tRUN NAME            \tSTATUS\tREVISION ID\tSESSION ID                          \tCOMMAND                                                  \r\n",
      "2022-09-21 08:41:46\t2.6s    \tloving_hilbert      \tERR   \t548d90ac1a \t0043a0d6-2c18-4262-aa47-67267d47047f\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616       \r\n",
      "2022-09-21 08:50:20\t11m 12s \tthirsty_kilby       \tERR   \t548d90ac1a \t3b816d86-bd01-4286-af07-fd57d5b63e8c\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616       \r\n",
      "2022-09-28 05:48:13\t28m 59s \tspecial_euclid      \tERR   \t548d90ac1a \t330a73f9-c98d-4d55-b85e-09cf9defa9c0\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616       \r\n",
      "2022-09-28 06:40:19\t30m 46s \ttiny_chandrasekhar  \tERR   \t08b5f112e9 \ta2212116-b3b4-4076-8e90-b3138f6ad235\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616        \r\n",
      "2022-09-28 07:20:19\t33m     \tclever_cantor       \tERR   \t858af7ee5f \t44a0ac2e-b5a5-4730-b8a9-29d8dace9483\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616        \r\n",
      "2022-09-28 07:56:09\t32m 33s \tgloomy_darwin       \tERR   \t1023da4922 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616        \r\n",
      "2022-09-28 08:44:47\t2.4s    \tfurious_darwin      \tERR   \t8c0f452f9c \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616 -resume\r\n",
      "2022-09-28 08:46:53\t42m 45s \tclever_chandrasekhar\tOK    \t3a80d28216 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616 -resume\r\n"
     ]
    }
   ],
   "source": [
    "!nextflow log\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "45574321",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TIMESTAMP          \tDURATION\tRUN NAME                \tSTATUS\tREVISION ID\tSESSION ID                          \tCOMMAND                                                                                                                                                                              \r\n",
      "2022-09-21 08:41:46\t2.6s    \tloving_hilbert          \tERR   \t548d90ac1a \t0043a0d6-2c18-4262-aa47-67267d47047f\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616                                                                                                                                   \r\n",
      "2022-09-21 08:50:20\t11m 12s \tthirsty_kilby           \tERR   \t548d90ac1a \t3b816d86-bd01-4286-af07-fd57d5b63e8c\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616                                                                                                                                   \r\n",
      "2022-09-28 05:48:13\t28m 59s \tspecial_euclid          \tERR   \t548d90ac1a \t330a73f9-c98d-4d55-b85e-09cf9defa9c0\tnextflow run ./hw1.nf --SRA SRR6410603, SRR6410616                                                                                                                                   \r\n",
      "2022-09-28 06:40:19\t30m 46s \ttiny_chandrasekhar      \tERR   \t08b5f112e9 \ta2212116-b3b4-4076-8e90-b3138f6ad235\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616                                                                                                                                    \r\n",
      "2022-09-28 07:20:19\t33m     \tclever_cantor           \tERR   \t858af7ee5f \t44a0ac2e-b5a5-4730-b8a9-29d8dace9483\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616                                                                                                                                    \r\n",
      "2022-09-28 07:56:09\t32m 33s \tgloomy_darwin           \tERR   \t1023da4922 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616                                                                                                                                    \r\n",
      "2022-09-28 08:44:47\t2.4s    \tfurious_darwin          \tERR   \t8c0f452f9c \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616 -resume                                                                                                                            \r\n",
      "2022-09-28 08:46:53\t42m 45s \tclever_chandrasekhar    \tOK    \t3a80d28216 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./hw1.nf --SRA SRR6410603,SRR6410616 -resume                                                                                                                            \r\n",
      "2022-10-08 10:50:35\t-       \ttrusting_yalow          \t-     \t0187cd0672 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:50:53\t-       \tgloomy_sinoussi         \t-     \t592d6446b0 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:51:55\t-       \tdeadly_leavitt          \t-     \t592d6446b0 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:52:39\t-       \tmaniac_leakey           \t-     \t592d6446b0 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:54:49\t-       \thigh_hypatia            \t-     \tb6f9365e74 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:58:03\t-       \tsick_bhaskara           \t-     \t467cca0fd1 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 10:58:56\t3.2s    \tadoring_noether         \tERR   \t28eada4454 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:02:09\t2.6s    \ttiny_jones              \tERR   \tb7a57030cb \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:05:02\t3.3s    \tclever_banach           \tERR   \t9e3497b320 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:05:36\t2.6s    \tangry_meucci            \tERR   \t9e3497b320 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:06:56\t2.7s    \tchaotic_shannon         \tERR   \td4c8205c9b \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:09:35\t2.7s    \tsilly_jennings          \tERR   \t6905d4564f \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:10:43\t3.1s    \tcondescending_heisenberg\tERR   \t5fc7c7c1fb \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index ./index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615                                   \r\n",
      "2022-10-08 11:17:49\t12m 29s \tfabulous_colden         \tERR   \tdd06836b34 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba2982615\r\n",
      "2022-10-08 11:44:55\t2.8s    \tloving_swartz           \tERR   \t68d8a99f71 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 11:58:38\t3.6s    \tadmiring_ritchie        \tERR   \t0a969de744 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 12:00:59\t-       \tadmiring_lamarck        \t-     \t8e4c6515eb \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 12:01:28\t11m 32s \tmad_cajal               \tERR   \t9c4a157470 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 12:19:22\t11m 12s \ttiny_bose               \tERR   \tbf6e7488ca \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 13:04:14\t21.1s   \tfestering_bohr          \tERR   \t0daf5b8331 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 13:05:15\t12m 31s \tspecial_allen           \tERR   \t0daf5b8331 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 13:39:00\t11m 56s \tjolly_elion             \tERR   \td05089f747 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 13:54:05\t11m 54s \tlonely_nobel            \tERR   \ta75321eb28 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 14:19:45\t12m 7s  \tnostalgic_pasteur       \tERR   \t00d5f45148 \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume                                     \r\n",
      "2022-10-08 14:36:50\t12m 55s \tlethal_majorana         \tERR   \t4c94222f1f \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba29     \r\n",
      "2022-10-08 15:11:31\t14m 49s \ttender_colden           \tERR   \tedffe2460c \t5c82254a-d3bd-4604-bfbc-118ba2982615\tnextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba29     \r\n"
     ]
    }
   ],
   "source": [
    "! nextflow log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "96bd1396",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "N E X T F L O W  ~  version 22.04.5\n",
      "Launching `./rna_quant.nf` [scruffy_boyd] DSL2 - revision: 17ebcd8533\n",
      "\n",
      "  Q U A L I T Y   C O N T R O L  \n",
      "=================================\n",
      "SRA number         : [SRR6410603, SRR6410616]\n",
      "Results location   : /mnt/data/prokaryota/safonnikova/fbb/res_1/results/\n",
      "Transcriptome index : /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx\n",
      "[-        ] process > DownloadFastQ -\u001b[K\n",
      "[-        ] process > QC            -\u001b[K\n",
      "[-        ] process > MultiQC       -\u001b[K\n",
      "\u001b[4A\n",
      "[be/af4fcc] process > DownloadFastQ (2) [100%] 2 of 2, cached: 2\u001b[K\n",
      "[-        ] process > QC                -\u001b[K\n",
      "[-        ] process > MultiQC           -\u001b[K\n",
      "[-        ] process > Kallisto          -\u001b[K\n",
      "\u001b[5A\n",
      "executor >  local (2)\u001b[K\n",
      "[be/af4fcc] process > DownloadFastQ (2) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[e3/2460f1] process > QC (2)            [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e9/35bb30] process > Kallisto (1)      [  0%] 0 of 2\u001b[K\n",
      "\u001b[6A\n",
      "executor >  local (2)\u001b[K\n",
      "[be/af4fcc] process > DownloadFastQ (2) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[e3/2460f1] process > QC (2)            [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e9/35bb30] process > Kallisto (1)      [  0%] 0 of 2\u001b[K\n",
      "\u001b[6A\n",
      "executor >  local (2)\u001b[K\n",
      "[be/af4fcc] process > DownloadFastQ (2) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[e3/2460f1] process > QC (2)            [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[55/175d25] process > Kallisto (2)      [ 50%] 1 of 2\u001b[K\n",
      "\u001b[6A\n",
      "executor >  local (2)\u001b[K\n",
      "[be/af4fcc] process > DownloadFastQ (2) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[e3/2460f1] process > QC (2)            [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
      "[48/7b0202] process > MultiQC           [100%] 1 of 1, cached: 1 ✔\u001b[K\n",
      "[e9/35bb30] process > Kallisto (1)      [100%] 2 of 2 ✔\u001b[K\n",
      "\u001b[32;1mCompleted at: 08-Oct-2022 16:38:25\n",
      "Duration    : 19m 59s\n",
      "CPU hours   : 1.9 (71.1% cached)\n",
      "Succeeded   : 2\n",
      "Cached      : 5\n",
      "\u001b[22;39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!./nextflow run ./rna_quant.nf --SRA SRR6410603,SRR6410616 --index /mnt/data/prokaryota/safonnikova/fbb/index/Homo_sapiens_GRCh38_index.idx -resume 5c82254a-d3bd-4604-bfbc-118ba29 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5fc1de65",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: .command.out: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "! cd ./work/db/d4f5ed1a8a8d15c0e86ef70bd5d52f/\n",
    "! cat .command.out"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
